### PR TITLE
feat: Jcv true injectable language

### PR DESCRIPTION
**Pull request by leomillon**
Sorry, this is a huge one...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,50 +6,91 @@ IntelliJ IDEA plugin for an enhanced coding experience on JCV based projects.
...
-designed to cover the most common needs when validating data with non-predictable values.
+### Your custom ones
 
-[JCV-DB](https://github.com/ekino/jcv-db) reuses these validators and also defines a list of its own 
-(listed [here](https://github.com/ekino/jcv-db/wiki/Validators)) that are specific to a database-oriented usage.
+You can define your own custom validator definitions by adding a `.jcvdefinitions.json` file in the project root directory or just let the plugin do it for you:
+
+![](./screenshots/jcv-demo-custom_validator_definition.gif)
+
+Here is and example definition you can customize:
```
Comment (mariebnst) [inline README.md:48]: *an

File: README.md [hunk 2]
```diff
@@ -6,50 +6,91 @@ IntelliJ IDEA plugin for an enhanced coding experience on JCV based projects.
...
+}
+```
+
+## Replacements suggestions
+
+The plugin can find raw JSON value that can be replaced by official matching validators.
+
+![](./screenshots/jcv-demo-replacements_suggestions.png)
+
+Note that it also support multi-caret replacement with matching validators for all JSON values find at each caret.
```
Comment (mariebnst) [inline README.md:74]: *supports

File: src/main/kotlin/com/ekino/oss/jcv/idea_plugin/language/Jcv.flex [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    {PARAMETER_CHARACTER} { yybegin(PARAMETER); }
+}
+
+<PARAMETER> {
+    {PARAMETER_SEPARATOR} { yypushback(1); yybegin(PARAMETERS); return JcvTypes.PARAMETER; }
+    {VALIDATOR_END_MARKER} { yypushback(2); yybegin(VALIDATOR); return JcvTypes.PARAMETER; }
+    {LineTerminator} { yypushback(1); yybegin(YYINITIAL); return JcvTypes.PARAMETER; }
+    <<EOF>> { yybegin(YYINITIAL); return JcvTypes.PARAMETER; }
+    {PARAMETER_CHARACTER} {}
+}
```
Comment (mariebnst) [inline src/main/kotlin/com/ekino/oss/jcv/idea_plugin/language/Jcv.flex:60]: 👏 👏 👏

File: src/main/kotlin/com/ekino/oss/jcv/idea_plugin/language/JcvLexerAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.ekino.oss.jcv.idea_plugin.language
+
+import com.intellij.lexer.FlexAdapter
+
+class JcvLexerAdapter : FlexAdapter(JcvLexer(null))
+
+
+private val value = listOf("val1","value",)
```
Comment (mariebnst) [inline src/main/kotlin/com/ekino/oss/jcv/idea_plugin/language/JcvLexerAdapter.kt:8]: Is there a use to this line ? It looks weird 😛

Comment (leomillon) [reply src/main/kotlin/com/ekino/oss/jcv/idea_plugin/language/JcvLexerAdapter.kt:8]: Wow good catch!

**Review by mariebnst — APPROVED** [review]

**Pull request merged**